docs/preparing_recipes_for_conan2.md

```diff
@@ -0,0 +1,368 @@
+    name = "ktx"
+    ...
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "KTX" 
+        ...
+        # FIXME: Remove the libktx component in Conan 2.0, this is just needed for
+        # compatibility with legacy generators
+        self.cpp_info.components["libktx"].names["cmake_find_package"] = "ktx"
+        ...
+        self.cpp_info.components["libktx"].libs = ["ktx"]
+        self.cpp_info.components["libktx"].defines = [
+            "KTX_FEATURE_KTX1", "KTX_FEATURE_KTX2", "KTX_FEATURE_WRITE"
+        ]
+
+        # Set the root cpp_info target name as KTX::ktx
+        # it will inherit all the libbrary properties (libs, defines, etc.) 
+        # from the defined component. In Conan 2.0 the component should be removed
+        # and those properties should be added to the root cpp_info instead
+        self.cpp_info.set_property("cmake_target_name", "KTX::ktx")
+        ...
+```
+
+* **Use build modules to create aliases with arbitray names for targets**. Similar to the
+  previous example, some recipes use a build module with an alias to set an arbitray
+  target name. Let's see the example of the [tensorflow-lite
+  recipe](https://github.com/conan-io/conan-center-index/blob/03b24bf128cbf15d23ed988b8d8ca0c0ba87d307/recipes/tensorflow-lite/all/conanfile.py),
+  that uses this workaround to define a ``tensorflow::tensorflowlite`` target.
+  
+```python
+class TensorflowLiteConan(ConanFile):
+    name = "tensorflow-lite"
+    ...
+
+    def package_info(self):
+        # generate the target tensorflowlite::tensorflowlite
+        self.cpp_info.names["cmake_find_package"] = "tensorflowlite"
+        self.cpp_info.filenames["cmake_find_package"] = "tensorflowlite"
+        # this build module defines an alias tensorflow::tensorflowlite to the tensorflowlite::tensorflowlite generated target
+        self.cpp_info.build_modules["cmake_find_package"] = [os.path.join(self._module_subfolder, self._module_file)]
+        ...
+```
+
+To translate this information to the new model, just check which aliases are defined in the
+build modules and define those for the new model. In this case it should be enough with
+adding the ``tensorflow::tensorflowlite`` target with ``cmake_target_name`` to the root
+cpp_info (besides the ``cmake_file_name``property).
+
+```python
+class TensorflowLiteConan(ConanFile):
+    name = "tensorflow-lite"
+    ...
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "tensorflowlite"
+        self.cpp_info.filenames["cmake_find_package"] = "tensorflowlite"
+        self.cpp_info.build_modules["cmake_find_package"] = [os.path.join(self._module_subfolder, self._module_file)]
+
+        # set the tensorflowlite::tensorflowlite target name directly for CMakeDeps with no need for aliases
+        self.cpp_info.set_property("cmake_target_name", "tensorflow::tensorflowlite")
+        self.cpp_info.set_property("cmake_file_name", "tensorflowlite")
```

Comment:
Is the target `tensorflowlite::tensorflowlite` available or only the `tensorflow::tensorflowlite` is now available? I mean, is it actually possible to create an alias without _overriding_ the original target name?

Reply:
Maybe in this case we want to use the `cmake_target_aliases` property instead of overwritting the target?

Reply:
but alias is easy: create another component depending on the other.